neo/rawio/maxwellrawio.py

```diff
@@ -0,0 +1,221 @@
+"""
+Class for reading data from maxwell biosystem device:
+  * MaxOne
+  * MaxTwo
+
+https://www.mxwbio.com/resources/mea/
+
+The implementation is a mix between:
+  * the implementation in spikeextractors
+     https://github.com/SpikeInterface/spikeextractors/blob/master/spikeextractors/extractors/maxwellextractors/maxwellextractors.py
+ * the implementation in spyking-circus
+    https://github.com/spyking-circus/spyking-circus/blob/master/circus/files/maxwell.py
+
+The implementation do not handle spike at the moment.
+
+For maxtwo device, each well will be a different signal stream.
+
+Author : Samuel Garcia, Alessio Buccino, Pierre Yger
+"""
+import os
+from pathlib import Path
+import platform
+from urllib.request import urlopen
+
+from .baserawio import (BaseRawIO, _signal_channel_dtype, _signal_stream_dtype,
+                _spike_channel_dtype, _event_channel_dtype)
+
+import numpy as np
+
+try:
+    import h5py
+    HAVE_H5 = True
+except ImportError:
+    HAVE_H5 = False
+
+
+class MaxwellRawIO(BaseRawIO):
+    """
+    Class for reading MaxOne or MaxTwo files.
+    """
+    extensions = ['h5']
+    rawmode = 'one-file'
+
+    def __init__(self, filename='', rec_name=None):
+        BaseRawIO.__init__(self)
+        self.filename = filename
+        self.rec_name = rec_name
+
+    def _source_name(self):
+        return self.filename
+
+    def _parse_header(self):
+        try:
+            import MEArec as mr
+            HAVE_MEAREC = True
+        except ImportError:
+            HAVE_MEAREC = False
+        assert HAVE_H5, 'h5py is not installed'
+
+        h5 = h5py.File(self.filename, mode='r')
+        self.h5_file = h5
+        version = h5['version'][0].decode()
+
+        # create signal stream
+        # one stream per well
+        signal_streams = []
+        if int(version) == 20160704:
+            self._old_format = True
+            signal_streams.append(('well000', 'well000'))
+        elif int(version) > 20160704:
+            # multi stream stream (one well is one stream)
+            self._old_format = False
+            stream_ids = list(h5['wells'].keys())
+            for stream_id in stream_ids:
+                rec_names = list(h5['wells'][stream_id].keys())
+                if len(rec_names) > 1:
+                    if self.rec_name is None:
+                        raise ValueError('several recording need select with rec_name="rec0000"'
+                            f'\nPossible rec_name {rec_names}')
+                else:
+                    self.rec_name = rec_names[0]
+                signal_streams.append((stream_id, stream_id))
+        signal_streams = np.array(signal_streams, dtype=_signal_stream_dtype)
+
+        # create signal channels
+        max_sig_length = 0
+        self._signals = {}
+        sig_channels = []
+        for stream_id in signal_streams['id']:
+            if int(version) == 20160704:
```

Comment:
same here: are you sure no files exist with older versions?

Reply:
@samuelgarcia wdyt?

Reply:
I have add a raise NotImplemented error